internals/plan/plan.go

```diff
@@ -184,7 +188,33 @@ func (s *Service) Merge(other *Service) {
 	if other.BackoffLimit.IsSet {
 		s.BackoffLimit = other.BackoffLimit
 	}
-	s.LogTargets = appendUnique(s.LogTargets, other.LogTargets...)
+
+	// Merge log targets
+	if other.LogTargetsReplace != nil {
+		if other.LogTargetsReplace.Default {
+			s.LogTargets = nil
+		} else {
+			s.LogTargets = other.LogTargetsReplace.Targets
+		}
+	} else if s.LogTargets == nil {
+		s.LogTargets = copyStrings(other.LogTargets)
+	} else {
+		s.LogTargets = appendUnique(s.LogTargets, other.LogTargets...)
+	}
+}
+
+// copyStrings returns a copy of the provided []string, in such a way that:
+//   - copyStrings(nil) returns nil;
+//   - copyStrings([]string{}) returns []string{}.
+//
+// TODO: move this function into canonical/x-go/strutil
```

Comment:
I think TODOs that are just "maybe we'll do this sometime" just add noise. When we want to do it, let's just do it, but otherwise it's fine as is.